src/Raven.Server/Config/Categories/ConfigurationCategory.cs

```diff
@@ -52,34 +52,46 @@ string GetValue(IConfiguration cfg, string name)
                     : null;
             }
 
-            string GetConfiguration(IConfiguration cfg, string name)
+            string GetConfigurationValue(IConfiguration cfg, string keyName, out bool keyExistsInConfiguration)
             {
-                if (cfg == null || name == null)
+                keyExistsInConfiguration = false;
+
+                if (cfg == null || keyName == null)
                     return null;
 
-                var val = GetValue(cfg, name);
+                // This check is needed because cfg.GetSection(keyName) returns null even if key does Not exist in configuration!
+                var keyNames = cfg.AsEnumerable().ToDictionary(x => x.Key, x => x.Value);
+                if (keyNames.ContainsKey(keyName))
+                {
+                    keyExistsInConfiguration = true;
+                }
+
+                var val = GetValue(cfg, keyName);
                 if (val != null)
                     return val;
 
-                var sb = new StringBuilder(name);
+                var sb = new StringBuilder(keyName);
 
-                var lastPeriod = name.LastIndexOf('.');
+                var lastPeriod = keyName.LastIndexOf('.');
                 while (lastPeriod != -1)
                 {
                     sb[lastPeriod] = ':';
                     var tmpName = sb.ToString();
                     val = GetValue(cfg, tmpName);
                     if (val != null)
                         return val;
-                    lastPeriod = name.LastIndexOf('.', lastPeriod - 1);
+                    lastPeriod = keyName.LastIndexOf('.', lastPeriod - 1);
                 }
 
                 return null;
             }
 
-            Initialize(key => new SettingValue(GetConfiguration(settings, key), GetConfiguration(serverWideSettings, key)),
-                serverWideSettings?[RavenConfiguration.GetKey(x => x.Core.DataDirectory)], type, resourceName,
-                throwIfThereIsNoSetMethod: true);
+            bool keyExistsInDatabaseRecord, keyExistsInServerSettings;
+            
+            Initialize(
+                key => new SettingValue(GetConfigurationValue(settings, key, out keyExistsInDatabaseRecord), keyExistsInDatabaseRecord,
```

Comment:
FYI, if you do not use variable, you can use `out _` instead

Reply:
but it is used as the second and fourth parameters